### PR TITLE
add tidb failover limit

### LINKED ISSUE #141: Tidb auto failover always add member when pumb failed
current the tidb auto-failover always add member if the pumb failed , but the tidb is still unvailable.
```
demo-tidb-0                       1/1       Running            2          4h
demo-tidb-1                       1/1       Running            2          4h
demo-tidb-10                      1/1       Running            2          2h
demo-tidb-11                      1/1       Running            2          2h
demo-tidb-12                      1/1       Running            2          2h
demo-tidb-13                      1/1       Running            2          2h
demo-tidb-14                      1/1       Running            2          2h
demo-tidb-15                      1/1       Running            2          2h
demo-tidb-16                      1/1       Running            2          2h
demo-tidb-17                      1/1       Running            2          2h
demo-tidb-18                      1/1       Running            2          2h
demo-tidb-19                      1/1       Running            2          2h
demo-tidb-2                       1/1       Running            2          3h
demo-tidb-20                      1/1       Running            2          1h
demo-tidb-21                      1/1       Running            2          1h
demo-tidb-22                      1/1       Running            2          1h
demo-tidb-23                      1/1       Running            2          1h
demo-tidb-24                      1/1       Running            2          1h
demo-tidb-25                      1/1       Running            2          1h
demo-tidb-26                      1/1       Running            2          1h
demo-tidb-27                      1/1       Running            0          20m
demo-tidb-28                      1/1       Running            0          20m
demo-tidb-29                      1/1       Running            0          15m
demo-tidb-3                       1/1       Running            2          3h
demo-tidb-30                      1/1       Running            0          15m
demo-tidb-31                      1/1       Running            0          10m
demo-tidb-32                      1/1       Running            1          10
```
operator need to identify the situation where tidb is unavailable due to pumb failure and in the situation auto-failover feature will stop.
in the other hand, tidb/tikv/pd should to limit auto-failover number, avoid unrestricted increase of members

### PR DESCRIPTION
**Pull request by xiaojingchen**
this PR add tidb's faileover limit to avoiding tidb failover out of control caused by non-tidb's own problems .
fixes #141

### CODE REVIEW

File: pkg/manager/member/tidb_failover.go [hunk 1]
```diff
@@ -35,6 +36,17 @@ func (tf *tidbFailover) Failover(tc *v1alpha1.TidbCluster) error {
 		tc.Status.TiDB.FailureMembers = map[string]v1alpha1.TiDBFailureMember{}
 	}
 
+	for _, tidbMember := range tc.Status.TiDB.Members {
```
Comment (weekface) [inline pkg/manager/member/tidb_failover.go]: These logics are recoveries, not failover. Recovery is here: https://github.com/weekface/tidb-operator/blob/115fe19f348e8546f4e10705e4729fd0aa9236d0/pkg/manager/member/tidb_member_manager.go#L150-L149

File: pkg/manager/member/tidb_failover_test.go [hunk 2]
```diff
@@ -112,6 +112,48 @@ func TestFakeTiDBFailoverFailover(t *testing.T) {
...
+					"failover-tidb-1": {
+						PodName: "failover-tidb-1",
+					},
+				}
+			},
+			errExpectFn: func(t *GomegaWithT, err error) {
+				t.Expect(err).NotTo(HaveOccurred())
+			},
+			expectFn: func(t *GomegaWithT, tc *v1alpha1.TidbCluster) {
+				t.Expect(len(tc.Status.TiDB.FailureMembers)).To(Equal(3))
```
Comment (weekface) [inline pkg/manager/member/tidb_failover_test.go]: The `Failover` method adds only one `failureMember` at a time, so this spec is useless.

File: charts/tidb-cluster/values.yaml [hunk 3]
```diff
@@ -136,6 +136,7 @@ tidb:
   #   operator: Equal
   #   value: tidb
   #   effect: "NoSchedule"
+  maxFailoverCount: 3
```
Comment (weekface) [inline charts/tidb-cluster/values.yaml]: Change `images/tidb-operator-e2e/tidb-cluster-values.yaml` also.

**Comment by tennix** [comment]
Why define this variable in `TidbCluster` spec? We already have other failover settings in tidb-operator

**Comment by xiaojingchen** [comment]
@tennix  Cause it is not an attribute of tidb-operator and this variable is customizable in each cluster. 
the same reason the other failover settings may be move to `TidbCluster` too

**Comment by weekface** [comment]
@xiaojingchen please address all the comments.

File: pkg/manager/member/tidb_failover_test.go (hunk 2 above)
Comment (xiaojingchen) [reply pkg/manager/member/tidb_failover_test.go]: have resolved

File: charts/tidb-cluster/values.yaml (hunk 3 above)
Comment (xiaojingchen) [reply charts/tidb-cluster/values.yaml]: have resolved

File: pkg/manager/member/tidb_failover.go (hunk 1 above)
Comment (xiaojingchen) [reply pkg/manager/member/tidb_failover.go]: this logic is hard to add to recovery function, cause recovery currently only happen when all members are running.

**Pull request merged**